lib/rubocop/cop/rails/custom_validator.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop suggests extracting custom validator classes from format validations.
+      #
+      # @example
+      #   # bad
+      #   class User < ApplicationRecord
+      #     validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i }
+      #   end
+      #
+      #   # good
+      #   class EmailValidator < ActiveModel::EachValidator
+      #     def validate_each(record, attribute, value)
+      #       unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+      #         record.errors[attribute] << (options[:message] || 'is not a valid email')
+      #       end
+      #     end
+      #   end
```

Comment:
I'm wondering that implementing the email format in an application is the first choice. For example, developer can consider using the following gem (or similar gems):
https://github.com/validates-email-format-of/validates_email_format_of

Reply:
If developer decides to use external gem, then there won't be an offense, because no `validates ... format: ...` will be used for that case.

Should I change something in the implementation or do you have any suggestions?